### PR TITLE
Add control API for shard evacuation

### LINKED ISSUE #1086: Implement data evacuation command in Control API
Blocked by https://github.com/nspcc-dev/neofs-node/issues/1085

Provide Control API methods to trigger shard data dump and restore.
These methods should be available from CLI

```
$ neofs-cli control shards dump --id [..] --path [..]
$ neofs-cli control shards restore --id [..] --path [..]
```

### PR DESCRIPTION
**Pull request by fyrchik**
Depends on #1089 .
Close #1086.

### CODE REVIEW

File: cmd/neofs-cli/modules/dump.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+var dumpShardCmd = &cobra.Command{
+	Use:   "dump",
+	Short: "Dump objects from shard",
+	Long:  "Dump objects from shard to a file",
+	Run:   dumpShard,
+}
+
+func dumpShard(cmd *cobra.Command, _ []string) {
+	key, err := getKey()
```
Comment (carpawell) [inline cmd/neofs-cli/modules/dump.go:24]: i think we can use here (and in all `shard` section) `getKeyNoGenerate`

File: cmd/neofs-cli/modules/dump.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+	exitOnErr(cmd, err)
+
+	body := new(control.DumpShardRequest_Body)
+
+	rawID, err := base58.Decode(shardID)
+	exitOnErr(cmd, errf("incorrect shard ID encoding: %w", err))
+	body.SetShardID(rawID)
+
+	p, _ := cmd.Flags().GetString(dumpFilepathFlag)
```
Comment (carpawell) [inline cmd/neofs-cli/modules/dump.go:33]: should we validate the path? (this is _path_, this is not empty, etc)

same for `restore`

**Review by carpawell — APPROVED** [review]

**Comment by carpawell** [comment]
#1089 has been merged, it can be rebased now

File: pkg/local_object_storage/engine/dump.go [hunk 3]
```diff
@@ -0,0 +1,19 @@
+package engine
+
+import "github.com/nspcc-dev/neofs-node/pkg/local_object_storage/shard"
+
+// DumpShard sets mode of the shard with provided identifier.
```
Comment (cthulhu-rider) [inline pkg/local_object_storage/engine/dump.go:5]: `dumps objects`? Same for restore

**Review by cthulhu-rider — APPROVED** [review]

File: cmd/neofs-cli/modules/dump.go (hunk 2 above)
Comment (alexvanin) [reply cmd/neofs-cli/modules/dump.go:33]: This path should exist on storage node side, which can be remote node, thus path can't be checked.

**Review by alexvanin — APPROVED** [review]
LGTM, besides https://github.com/nspcc-dev/neofs-node/pull/1105/files#r792747876
DNT.

File: cmd/neofs-cli/modules/dump.go (hunk 2 above)
Comment (carpawell) [reply cmd/neofs-cli/modules/dump.go:33]: i just meant to check the basic rules: go lib must be able to parse it correctly; path is not `""`; etc

as i see, nothing would stop cli from sending RPC call to the node that asks to dump everything to `some/awesome\path_on-my machine.exe` path

File: cmd/neofs-cli/modules/dump.go (hunk 1 above)
Comment (fyrchik) [reply cmd/neofs-cli/modules/dump.go:24]: Fixed.

File: pkg/local_object_storage/engine/dump.go (hunk 3 above)
Comment (fyrchik) [reply pkg/local_object_storage/engine/dump.go:5]: Fixed.

File: pkg/services/control/service.proto [hunk 4]
```diff
@@ -201,3 +207,72 @@ message SetShardModeResponse {
...
+        bytes shard_ID = 1;
+
+        // Path to the output.
+        string filepath = 2;
+
+        // Flag indicating whether object read errors should be ignored.
+        bool ignore_errors = 3;
+    }
+
+    // Body of set shard mode request message.
```
[the remark this answers is not in the document]
Comment (fyrchik) [reply pkg/services/control/service.proto:225]: Fixed.

File: pkg/services/control/service.proto [hunk 5]
```diff
@@ -201,3 +207,72 @@ message SetShardModeResponse {
...
+    message Body {
+    }
+
+    // Body of set shard mode response message.
+    Body body = 1;
+
+    // Body signature.
+    Signature signature = 2;
+}
+
```
[the remark this answers is not in the document]
Comment (fyrchik) [reply pkg/services/control/service.proto:244]: Fixed.

**Pull request merged**